snuba/subscriptions/data.py

```diff
@@ -74,20 +76,56 @@ def from_string(cls, value: str) -> SubscriptionIdentifier:
         return cls(PartitionId(int(partition)), UUID(uuid))
 
 
-@dataclass(frozen=True)
-class SubscriptionData:
-    """
-    Represents the state of a subscription.
-    """
-
+@dataclass(frozen=True, kw_only=True)
+class _SubscriptionData:
     project_id: int
     resolution_sec: int
     time_window_sec: int
     entity: Entity
-    query: str
     metadata: Mapping[str, Any]
     tenant_ids: Mapping[str, Any] = field(default_factory=lambda: dict())
 
+
+TRequest = TypeVar("TRequest")
```

Comment:
Generic return type for `build_request` since it's going to be different for RPCs